### ISSUE TITLE
Debug package fails to link

### ISSUE THREAD

**Issue by helmesjo** [helmesjo/conan-thrift#2]
**Error:**
```
../../.conan/data/thrift/0.11.0/helmesjo/stable/package/5e5bb7c019799aba7d94d3142313d1d24b1d3491/lib/libthriftnbd.a(TNonblockingServerSocket.cpp.o): In function `apache::thrift::transport::TNonblockingServerSocket::listen()':
/home/conan/.conan/data/thrift/0.11.0/helmesjo/stable/build/5e5bb7c019799aba7d94d3142313d1d24b1d3491/source_subfolder/lib/cpp/src/thrift/transport/TNonblockingServerSocket.cpp:196: undefined reference to `apache::thrift::transport::TGetAddrInfoWrapper::TGetAddrInfoWrapper(char const*, char const*, addrinfo const*)'
/home/conan/.conan/data/thrift/0.11.0/helmesjo/stable/build/5e5bb7c019799aba7d94d3142313d1d24b1d3491/source_subfolder/lib/cpp/src/thrift/transport/TNonblockingServerSocket.cpp:198: undefined reference to `apache::thrift::transport::TGetAddrInfoWrapper::init()'
/home/conan/.conan/data/thrift/0.11.0/helmesjo/stable/build/5e5bb7c019799aba7d94d3142313d1d24b1d3491/source_subfolder/lib/cpp/src/thrift/transport/TNonblockingServerSocket.cpp:208: undefined reference to `apache::thrift::transport::TGetAddrInfoWrapper::res()'
/home/conan/.conan/data/thrift/0.11.0/helmesjo/stable/build/5e5bb7c019799aba7d94d3142313d1d24b1d3491/source_subfolder/lib/cpp/src/thrift/transport/TNonblockingServerSocket.cpp:196: undefined reference to `apache::thrift::transport::TGetAddrInfoWrapper::~TGetAddrInfoWrapper()'
/home/conan/.conan/data/thrift/0.11.0/helmesjo/stable/build/5e5bb7c019799aba7d94d3142313d1d24b1d3491/source_subfolder/lib/cpp/src/thrift/transport/TNonblockingServerSocket.cpp:196: undefined reference to `apache::thrift::transport::TGetAddrInfoWrapper::~TGetAddrInfoWrapper()'
collect2: error: ld returned 1 exit status
```
**Reproduce:**

1. Add thrift to `conanfile.txt`: `thrift/0.11.0@helmesjo/stable`
2. Setup a target linking thrift:
    ```
    find_package(thrift REQUIRED)

    add_library( ${PROJECT_NAME}
        main.cpp
    )

    target_link_libraries(${PROJECT_NAME}
        PUBLIC thrift::thrift
    )
    ```
3. Setup a `TNonblockingServerSocket`:
```
    #include <thrift/transport/TNonblockingServerSocket.h>

    // ...

    auto serverTransport = std::make_shared<apache::thrift::transport::TNonblockingServerSocket>(port);

```
4. Install debug package and build target.

**Comment by helmesjo** [comment]
On a second thought, the [`test_package.cpp`](https://github.com/helmesjo/conan-thrift/blob/706c9256b1c2d377525aa0dd2e2377ee10313d74/test_package/test_package.cpp) builds, so steps to reproduce above might not be enough/the real issue...

**Comment by helmesjo** [comment]
Listing symbols in the static lib for linux (package [2b2db135a61925a58a00a01cc1a1b41802161630](https://bintray.com/helmesjo/public-conan/thrift%3Ahelmesjo#files/helmesjo%2Fthrift%2F0.11.0%2Fstable%2Fpackage%2F2b2db135a61925a58a00a01cc1a1b41802161630)) gives:

**libthrift.a**:
```
nm ./libthrift.a --demangle | grep TGetAddrInfoWrapper
00000000000004c0 T apache::thrift::transport::TGetAddrInfoWrapper::res()
0000000000000490 T apache::thrift::transport::TGetAddrInfoWrapper::init()
0000000000000460 T apache::thrift::transport::TGetAddrInfoWrapper::TGetAddrInfoWrapper(char const*, char const*, addrinfo const*)
0000000000000460 T apache::thrift::transport::TGetAddrInfoWrapper::TGetAddrInfoWrapper(char const*, char const*, addrinfo const*)
0000000000000050 T apache::thrift::transport::TGetAddrInfoWrapper::~TGetAddrInfoWrapper()
0000000000000020 T apache::thrift::transport::TGetAddrInfoWrapper::~TGetAddrInfoWrapper()
0000000000000020 T apache::thrift::transport::TGetAddrInfoWrapper::~TGetAddrInfoWrapper()
0000000000000000 V typeinfo for apache::thrift::transport::TGetAddrInfoWrapper
0000000000000000 V typeinfo name for apache::thrift::transport::TGetAddrInfoWrapper
0000000000000000 V vtable for apache::thrift::transport::TGetAddrInfoWrapper
```

**libthriftnb.a** (non blocking):
```
nm ./libthriftnb.a --demangle | grep TGetAddrInfoWrapper
                 U apache::thrift::transport::TGetAddrInfoWrapper::res()
                 U apache::thrift::transport::TGetAddrInfoWrapper::init()
                 U apache::thrift::transport::TGetAddrInfoWrapper::TGetAddrInfoWrapper(char const*, char const*, addrinfo const*)
                 U apache::thrift::transport::TGetAddrInfoWrapper::~TGetAddrInfoWrapper(
```

Notice **U** (undefined) in `libthriftnb.a`.

Is this expected since `libthriftnb.a` links `libthrift.a`? Is the above error because I'm not linking both to the third dependee..?

**Comment by helmesjo** [comment]
Or, am I linking them in the wrong order? Should be `-llibthriftnb -llibthrift`. **<---**

**Comment by helmesjo** [comment]
In short: The library that needs symbols must be first, then the library that resolves the symbols should come after.

**Comment by helmesjo** [comment]
Fixed in commit c8ee7b17e13dcddbfd81169dee72df6c341fc9e7.
